### PR TITLE
Load plan jurisdiction hierarchy from OpenSRP

### PR DESCRIPTION
**Pull request by moshthepitt**
This PR:

- refactors the TreeWalker component and its helpers to be able to work with location hierarchies from OpenSRP i.e. TreeWalker is now able to traverse the OpenSRP jurisdictions tree both with and without a location hierarchy
- refactors all the components of the plan assignment page to make them work with location hierarchy tree nodes
- adds some selectors to the hierarchies redux module
- add `getJurisdictions`; a convenience function that takes an array of jurisdiction ids and fetches them from OpenSRP
- cleans up wherever possible

Fix: #972

### CODE REVIEW

File: src/store/ducks/opensrp/hierarchies/index.ts [hunk 1]
```diff
@@ -297,6 +301,55 @@ export const getNodeById = () =>
...
+ * You should pass in rootJurisdictionId as '' when don't know it
+ *
+ * @param state - the store
+ * @param props -  the filterProps
+ */
+export const getRootByNodeId = () =>
+  createSelector(
+    [getTreesByIds, getNodeId, getRootJurisdictionId],
+    (trees, nodeId, rootJurisdictionId) => {
+      if (rootJurisdictionId && rootJurisdictionId !== '') {
```
Comment (KipSigei) [inline src/store/ducks/opensrp/hierarchies/index.ts:317]: can we use `!!rootJurisdictionId` here instead?

File: src/components/TreeWalker/helpers.ts [hunk 2]
```diff
@@ -83,78 +100,73 @@ export const getAncestors = async (
...
+export const getChildren = (
+  nodeId: string,
+  tree: TreeNode,
+  doFormat: boolean = true
+): Result<TreeNodeType[]> => {
+  let children: OpenSRPJurisdiction[] = [];
+  const nodeFromTree = tree.first(treeNode => treeNode.model.id === nodeId);
+  if (nodeFromTree && nodeFromTree.model.children) {
+    children =
+      doFormat === true
```
Comment (ciremusyoka) [inline src/components/TreeWalker/helpers.ts:118]: Do we need to equate a boolean?

File: src/components/TreeWalker/constants.ts [hunk 3]
```diff
@@ -1,5 +1,7 @@
 export const ACTIVE = 'Active';
+export const FEATURE = 'Feature';
 export const FIND_BY_ID = 'location/findByJurisdictionIds';
 export const FIND_BY_PROPERTIES = 'location/findByProperties';
 export const LOCATION = 'location';
+export const COULDNT_LOAD_CHILDREN = 'Could not load children';
 export const COULDNT_LOAD_PARENTS = 'Could not load parents';
```
Comment (ciremusyoka) [inline src/components/TreeWalker/constants.ts:7]: Should this strings be here or in the lang file?

File: src/store/ducks/opensrp/hierarchies/index.ts (hunk 1 above)
Comment (moshthepitt) [reply src/store/ducks/opensrp/hierarchies/index.ts]: Done

File: src/components/TreeWalker/helpers.ts (hunk 2 above)
Comment (moshthepitt) [reply src/components/TreeWalker/helpers.ts]: Removed

File: src/components/TreeWalker/constants.ts (hunk 3 above)
Comment (moshthepitt) [reply src/components/TreeWalker/constants.ts:7]: This component will be moved to OpenSRP web so it has its own constants module.  In this commit, we get the strings for display from the lang module: https://github.com/onaio/reveal-frontend/pull/1003/commits/7ee298e7766bb3da593ebe985be644bae1b47fa8

**Comment by Naima-Bashir** [comment]
How to test: https://github.com/onaio/reveal-frontend/issues/939

**Pull request merged**